### ISSUE TITLE
This is such a great mod to go with BDB! Any chance it will be updated?

### ISSUE THREAD

**Issue by Maxzhao1999** [jsolson/BDB-ProceduralFairings#2]

**Comment by jsolson** [comment]
I haven't used procedural fairings in a while but I could take a look. What needs updating?

**Comment by Gordon-Dry** [comment]
For me it works.

But see:
https://forum.kerbalspaceprogram.com/index.php?/topic/122020-14x-bluedog-design-bureau-stockalike-saturn-apollo-and-more-v142-i-numbered-this-update-before-ksp-updated-1feb2018/&do=findComment&comment=3377122

**Comment by Maxzhao1999** [comment]
There're some issues with procedural fairings and the patch creating two sets of side fairings. I deleted the pf_side.cfg and it seems to work better. Also the BDB fairing would be of the default size when first dropped into the editor but right clicking would apply the size restriction by technology level.

**Comment by Maxzhao1999** [comment]
Also, the PF version of the Daleth fairing has its bottom node missing

**Comment by jsolson** [comment]
The very old	Firespitter included might be the problem with pf_side.cfg. I wonder if this could be converted to use stock texture switching.

The delta fairing had an extra node for the interstage added since this was written. It needs to be accounted for.

**Comment by Maxzhao1999** [comment]
Yes I think the delta fairing is indeed caused by an added node. I hope it won't be hard to fix though!
As for the firespitter plugin, I've always used the most up to data ones. However I think implementing the stock texture switching would be great!

**Comment by jsolson** [comment]
Delta should work ok now.

>Also the BDB fairing would be of the default size when first dropped into the editor but right clicking would apply the size restriction by technology level.

I don't know that I can do anything about this.

It looks like Procedural Fairings took advantage of stock texture switching so there's not much point to pf_side anymore. That's all it did. How have the textures aged? Should we bother keeping them?

**Comment by Gordon-Dry** [comment]
GameData\Jso\Bdb\PF\bdb_pf_side.cfg
line 45
+PART[KzProcFairingSide3]:NEEDS[Firespitter]

**Comment by Maxzhao1999** [comment]
I don't think the fairing textures should be kept as it stands. Maybe a set of better quality texture implemented by the stock texture switching would be nice.

There is a lack of fairing texture resources in KSP mods and the few that once existed are extremely out of date.

**Comment by jsolson** [comment]
>+PART[KzProcFairingSide3]:NEEDS[Firespitter]

There's no `KzProcFairingSide3]` part to copy from, only 1 and 2. That line does nothing.

@Maxzhao1999 due to the nature of procedural parts in general it is difficult to make quality textures for them since the texture gets stretched and/or compressed. Anything other than a solid color suffers. No real artists have taken an interest in making them. The source files are here if you want to take a crack at it. I believe they are GIMP files.

**Comment by jsolson** [comment]
New release is up. I got the stock texture switching working. This will break saves since the extra BDB side parts are gone. The textures get added to the PF side parts now. The third Blunt fairing (Egg shaped with a flatter nose) is still there.

Firespitter is no longer needed.

**Comment by Gordon-Dry** [comment]
Hmm, still got issues with the Delta-K fairing tank - in VAB it sometimes showed the side nodes, then did not - in flight the fairing sides did not detach and the kraken hit me ...

I guess I should avoid the Delta K thingie...

**Comment by jsolson** [comment]
It has that extra bottom node it has for the interstage attachment that may render it incompatible... I don't recall having issues but I'll try to reproduce.
